core/src/main/java/org/apache/iceberg/MetricsUtil.java

```diff
@@ -172,21 +172,29 @@ public static MetricsModes.MetricsMode metricsMode(
               "Lower bound",
               DataFile.LOWER_BOUNDS,
               Types.NestedField::type,
-              (file, field) ->
-                  file.lowerBounds() == null
-                      ? null
-                      : Conversions.fromByteBuffer(
-                          field.type(), file.lowerBounds().get(field.fieldId()))),
+              (file, field) -> {
+                if (file.lowerBounds() == null) {
+                  return null;
+                }
+                Object value =
+                    Conversions.fromByteBuffer(
+                        field.type(), file.lowerBounds().get(field.fieldId()));
+                return (value instanceof java.util.UUID) ? value.toString() : value;
```

Comment:
I'm trying to follow the logic here, shouldn't this always return just bytes:

https://github.com/apache/iceberg/blob/76b2f875ff14f0ddeb80bfa63e819eeec3047975/api/src/main/java/org/apache/iceberg/types/Conversions.java#L172-L174

Reply:
Sorry, my brain is still on vacation, it looks good:

https://github.com/apache/iceberg/blob/76b2f875ff14f0ddeb80bfa63e819eeec3047975/api/src/main/java/org/apache/iceberg/types/Conversions.java#L170-L171